dashboard/app/models/levels/aichat.rb

```diff
@@ -0,0 +1,76 @@
+# == Schema Information
+#
+# Table name: levels
+#
+#  id                    :integer          not null, primary key
+#  game_id               :integer
+#  name                  :string(255)      not null
+#  created_at            :datetime
+#  updated_at            :datetime
+#  level_num             :string(255)
+#  ideal_level_source_id :bigint           unsigned
+#  user_id               :integer
+#  properties            :text(4294967295)
+#  type                  :string(255)
+#  md5                   :string(255)
+#  published             :boolean          default(FALSE), not null
+#  notes                 :text(65535)
+#  audit_log             :text(65535)
+#
+# Indexes
+#
+#  index_levels_on_game_id    (game_id)
+#  index_levels_on_level_num  (level_num)
+#  index_levels_on_name       (name)
+#
+
+# TODO: Update this level to account for the params we actually need.
+class Aichat < Level
+  serialized_attrs %w(
+    project_template_level_name
+    start_sources
+    hide_share_and_remix
+    is_project_level
+    submittable
+  )
+
+  def self.create_from_level_builder(params, level_params)
+    create!(
+      level_params.merge(
+        user: params[:user],
+        game: Game.aichat,
+        level_num: 'custom',
+        properties: {}
+      )
+    )
+  end
+
+  # Return an 'appOptions' hash derived from the level contents
+  def non_blockly_puzzle_level_options
+    options = Rails.cache.fetch("#{cache_key}/non_blockly_puzzle_level_options/v2") do
+      level_prop = {}
+
+      properties.keys.each do |dashboard|
+        apps_prop_name = dashboard.camelize(:lower)
+        # Select value from properties json
+        # Don't override existing valid (non-nil/empty) values
+        value = JSONValue.value(properties[dashboard].presence)
+        level_prop[apps_prop_name] = value unless value.nil? # make sure we convert false
+      end
+
+      level_prop['levelId'] = level_num
+
+      # We don't want this to be cached (as we only want it to be seen by authorized teachers), so
+      # set it to nil here and let other code put it in app_options
+      level_prop['teacherMarkdown'] = nil
+
+      # Don't set nil values
+      level_prop.reject! {|_, value| value.nil?}
+    end
+    options.freeze
+  end
+
+  def uses_lab2?
```

Comment:
since `uses_lab2` is true, we don't need a bunch of the boilerplate anymore. Everything gets loaded via `levels-lab2-main`, so you don't need the `aichat` version of that file, along with a bunch of other logic, you can see where I removed some of this for music [in this PR](https://github.com/code-dot-org/code-dot-org/pull/52667/files). It would also be worth merging the latest changes into this branch, it doesn't look like you have the changes from that linked pr here.

Reply:
Here's what I removed so far: https://github.com/code-dot-org/code-dot-org/pull/52538/commits/90cb70653e5282fb2567d2b1da099d3494283c1e